python/sqlflow_submitter/tensorflow/train_estimator.py

```diff
@@ -30,7 +30,7 @@ def estimator_train_and_save(estimator, model_params, save, is_pai, FLAGS,
     # NOTE(typhoonzero): checkpoints will be used by explaining (explaining BoostedTrees model
     # requires calling estimator.experimental_predict_with_explanations),
     # yet, predicting will use the saved model on OSS only.
-    if is_pai and FLAGS.task_index == 0:
+    if is_pai and FLAGS.task_index == 0 and FLAGS.job_name == "worker":
```

Comment:
This may also cause a distributed job fails because ps0 and worker0 maybe deleting the same checkpoint dir at the same time causing IO error.